xarray/core/options.py

```diff
@@ -18,6 +42,10 @@ class set_options(object):
       Default: ``80``.
     - ``arithmetic_join``: DataArray/Dataset alignment in binary operations.
       Default: ``'inner'``.
+    - ``io_engines``: List of backend data I/O engines to try in order when
+      saving/loading data without an explicit engine. Default: All the
+      installed engines on start up from this list: ['netcdf4', 'pydap',
+      'scipy', 'h5netcdf'].
```

Comment:
Thinking about the interface here again, would it make sense to use a mapping {file_format: engine} instead? That would be more flexible than a single list with priority order. For example: `{'netcdf3': 'scipy', 'netcdf4': 'h5netcdf', 'remote': 'pydap'}`.

Or perhaps these should all be separate options, e.g., `default_netcdf3_engine`, `default_netcdf4_engine` and `default_remote_engine`.

Reply:
I think how `_get_default_engine()` does it is sufficient for now.

Reply:
@jhamman @fmaussion any thoughts on API design here?

Reply:
I think I'm generally okay here with just providing a list. Since this option only affects the netcdf-like engines, it would be good to be clear about that in the doc string. Pynio, rasterio, and zarr are also io_engines but this option does not control their behavior. Perhaps the name of the option should be changed. Alternatively, an approach like @shoyer recommends would allow for an explicit mapping for any number of file types (not just netcdf).

Reply:
The current design allows adding other Python packages and associating them with the file format they support. Let me know what other packages you would like to see added.

For remote data access engines, I think nothing super-smart can be done because all of their URLs will start with `http`. Setting a default engine option (this PR) or using the `engine` keyword is inevitable.